packages/@aws-cdk/aws-codedeploy/lib/ecs/deployment-group.ts

```diff
@@ -55,8 +199,132 @@ export class EcsDeploymentGroup {
+  public readonly application: IEcsApplication;
+  public readonly deploymentGroupName: string;
+  public readonly deploymentGroupArn: string;
+  public readonly deploymentConfig: IEcsDeploymentConfig;
+  /**
+   * The role for this CodeDeploy to use when deploying to this deployment group.
+   */
+  public readonly role: iam.IRole;
+  private readonly alarms: cloudwatch.IAlarm[];
+
+  constructor(scope: Construct, id: string, props: EcsDeploymentGroupProps) {
+    super(scope, id, {
+      physicalName: props.deploymentGroupName,
+    });
+
+    this.application = props.application || new EcsApplication(this, 'Application');
+    this.alarms = props.alarms || [];
+
+    this.role = props.role || new iam.Role(this, 'ServiceRole', {
+      assumedBy: new iam.ServicePrincipal('codedeploy.amazonaws.com'),
+    });
+    this.role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AWSCodeDeployRoleForECS'));
+    this.deploymentConfig = props.deploymentConfig || EcsDeploymentConfig.ALL_AT_ONCE;
+
+    // Change ECS Service deployment controller to codedeploy
+    for (const service of props.services) {
+      const cfnService = service.node.defaultChild as ecs.CfnService;
+      cfnService.deploymentController = { type: ecs.DeploymentControllerType.CODE_DEPLOY };
+
+      // Remove revision from ECS Service task definition to allow Blue/Green deploys to work
+      cfnService.taskDefinition = service.taskDefinition.family;
+      service.node.addDependency(service.taskDefinition);
+    }
+
+    // create a testTrafficRoute if missing
+    let testTrafficRoute = props.testTrafficRoute;
+    if (props.prodTrafficRoute && !testTrafficRoute) {
+      const prodCfnTargetGroup = props.prodTrafficRoute.targetGroup.node.defaultChild as CfnTargetGroup;
+      const testCfnTargetGroup = new CfnTargetGroup(this, 'TestTargetGroup', {
+        healthCheckEnabled: prodCfnTargetGroup.healthCheckEnabled,
+        healthCheckIntervalSeconds: prodCfnTargetGroup.healthCheckIntervalSeconds,
+        healthCheckPath: prodCfnTargetGroup.healthCheckPath,
+        healthCheckPort: prodCfnTargetGroup.healthCheckPort,
+        healthCheckProtocol: prodCfnTargetGroup.healthCheckProtocol,
+        healthCheckTimeoutSeconds: prodCfnTargetGroup.healthCheckTimeoutSeconds,
+        healthyThresholdCount: prodCfnTargetGroup.healthyThresholdCount,
+        ipAddressType: prodCfnTargetGroup.ipAddressType,
+        matcher: prodCfnTargetGroup.matcher,
+        name: prodCfnTargetGroup.name,
+        port: prodCfnTargetGroup.port,
+        protocol: prodCfnTargetGroup.protocol,
+        protocolVersion: prodCfnTargetGroup.protocolVersion,
+        targetGroupAttributes: prodCfnTargetGroup.targetGroupAttributes,
+        targets: prodCfnTargetGroup.targets,
+        targetType: prodCfnTargetGroup.targetType,
+        unhealthyThresholdCount: prodCfnTargetGroup.unhealthyThresholdCount,
+        vpcId: prodCfnTargetGroup.vpcId,
+      });
```

Comment:
Hmmmm this feels doomed to not keep up with new attributes in the TargetGroup resource type and not copy everything over.  Is there any type of `copy` or `duplicate` method convention CDK we could add to the target group classes directly?

Reply:
I would also call these traffic group 1 vs 2, or blue vs green.  It's not accurate to call this a "test" target group as it will serve production traffic

Reply:
I was trying to auto-create the green TG as a clone of blue TG. More trouble than its worth - i've abandon that idea.